receiver/statsdreceiver/protocol/parser.go

```diff
@@ -20,7 +20,7 @@ import (
 
 // Parser is something that can map input StatsD strings to OTLP Metric representations.
 type Parser interface {
-	Initialize(enableMetricType bool, isMonotonicCounter bool, sendTimerHistogram []TimerHistogramMapping) error
+	Initialize(enableMetricType bool, isMonotonicCounter bool, sendTimerHistogram []TimerHistogramMapping, sourceAddresskey string) error
 	GetMetrics() pmetric.Metrics
-	Aggregate(line string) error
+	Aggregate(line, addr string) error
```

Comment:
If you're already aggregating by the client address, why don't we just add it to the context on the flush? Similar to how it's done in https://github.com/open-telemetry/opentelemetry-collector/blob/d5d818d8054c7c965af8a6ab28843cf1feb3c858/config/confighttp/clientinfohandler.go#L44

Reply:
It's possible, though on every flush interval it would produce N baches of of data (since context is attached to a batch) rather than a single batch we do now.
My hesitations were the change in behaviour and the slightly larger change it would incur (though looking back now, it doesn't look too big).

If you prefer, I can change it to do that.

Reply:
> It's possible, though on every flush interval it would produce N baches of of data (since context is attached to a batch) rather than a single batch we do now.

Where N is the number of different clients sending metrics with exactly the same attributes. Must be a pretty rare case right?

Reply:
I think using the batch processor, as generally recommended, should mitigate the multiple batches overhead. And this is a more performant approach to use with k8sattributes receiver than having to split the batches with gropbyattribute processor. Another benefit is that we don't need an extra configuration option and a resource attribute that won't be used after k8sattributes processor.

So I would rather go with putting the address in the context.

Reply:
Maybe we can make it optional tho?

Reply:
I've rewritten the PR to both store and emit metrics batched by the source address.